packages/aws-cdk/test/jest-setup-after-env.ts

```diff
@@ -0,0 +1,64 @@
+import * as fs from 'fs';
+import * as os from 'os';
+import * as path from 'path';
+
+/**
+ * Global test setup for Jest tests
+ *
+ * It's easy to accidentally write tests that interfere with each other by
+ * writing files to disk in the "current directory". To prevent this, the global
+ * test setup creates a directory in the temporary directory and chmods it to
+ * being non-writable. That way, whenever a test tries to write to the current
+ * directory, it will produce an error and we'll be able to find and fix the
+ * test.
+ *
+ * If you see `EACCES: permission denied`, you have a test that creates files
+ * in the current directory, and you should be sure to do it in a temporary
+ * directory that you clean up afterwards.
+ *
+ * ## Alternate approach
+ *
+ * I tried an approach where I would automatically try to create and clean up
+ * temp directories for every test, but it was introducing too many conflicts
+ * with existing test behavior (around specific ordering of temp directory
+ * creation and cleanup tasks that are already present) in many places that I
+ * didn't want to go and chase down.
+ *
+ */
+
+let tmpDir: string;
+let oldDir: string;
+
+beforeAll(() => {
+  tmpDir = path.join(os.tmpdir(), 'cdk-nonwritable');
+  if (!fs.existsSync(tmpDir)) {
```

Comment:
So If I manually change the permissions of this directory back to writable - suddenly I can locally write tests that write to this directory. 

Can we make this unique per execution?

Reply:
This is not a security measure that we have to make sure nobody can circumvent.

It's a measure to help you prevent mistakes.

Also, the test will still fail on the build server (unless you put a `chmod` in your code, but again why are you bypassing a measure that is helping you write more robust tests? This is not for security purposes!)

Reply:
Im not trying to tighten security, i'm trying to minimize scenarios where tests pass locally but fail on the server.

A contributor gets a weird permission error -> they don't know whats going on -> so they manually "fix" the issue by changing perms on the static directory. They write tests locally, all is well. But then, as you said, it fails on the build server. Thats not a great experience. What's the downside of having a unique dir per execution?

Reply:
How does having a unique dir per execution solve that scenario?

It prevents them fixing their local setup by `chmod`ding, but the build on the build server will still fail with the same error message. Doesn't seem like you've prevented user confusion.

Also, if someone gets as far as `chmod`ding a directory that's called `cdk-nonwritable`... 🙄 I don't know how to help. I can name it `cdk-nonwritable-on-purpose-do-not-chmod`, but I don't know if that makes it better.

Reply:
If they are able to get to that directory, it seems they could `grep` the codebase to get to this specific file and read the comment there as well.

Reply:
If there's a unique directory per execution, whatever manual permission change they do won't solve the problem.

> Because they changed permission of `/tmp/unique1/cdk-nonwritable`, but now the error is coming from `/tmp/unique2/cdk-nowritable`

And they will keep getting the error until they understand how to properly solve it. Not critical, take it or leave it.

Reply:
I did it another way